pkg/kubelet/kubelet.go

```diff
@@ -2238,6 +2238,11 @@ func (kl *Kubelet) ListenAndServePodResources() {
 
 // Delete the eligible dead container instances in a pod. Depending on the configuration, the latest dead containers may be kept around.
 func (kl *Kubelet) cleanUpContainersInPod(podID types.UID, exitedContainerID string) {
+	// Handles container resource immediately.
+	if err := kl.containerManager.GetInternalContainerLifecycle().PostStopContainer(exitedContainerID); err != nil {
```

Comment:
Hi again @hex108 :)

Was it not possible to just do this here?

```go
if err := kl.containerManager.InternalContainerLifecycle().PostStopContainer(exitedContainerID); err != nil {
```

instead of changing the `ContainerManager.InternalContainerLifecycle()` method to return a cached object?

Also, please add a quick code comment above the call to `PostStopContainer()` here that explains why this is being done here. That way, when someone reads this code in the future and sees there is another place in `deleteContainer()` that also calls `PostStopContainer`, they will not be tempted to remove it :)

Reply:
When calling `kl.containerManager.InternalContainerLifecycle().PostStopContainer(exitedContainerID)`, we want to clear status for the `exitedContainerId` in `InternalContainerLifecycle`.

https://github.com/kubernetes/kubernetes/blob/822a2de262372db4fb11aa4d69389c0cdbd869aa/pkg/kubelet/cm/internal_container_lifecycle.go#L63-L77

If we do not cache it, we could not remove status from cpumanager's own `InternalContainerLifecycle`, what we do is just clearing status from a new `InternalContainerLifecycle`, and it does not solve the problem.

> Also, please add a quick code comment above the call to PostStopContainer() here that explains why this is being done here. 

Addressed, PTAL.